addons/crm/tests/test_crm_lead_merge.py

```diff
@@ -294,3 +390,120 @@ def test_merge_method_dependencies(self):
+        }])
+
+        self.env['mail.message'].create([{
+            'author_id': self.contact_1.id,
+            'model': 'crm.lead',
+            'res_id': self.lead_w_partner_company.id,
+            'date': Datetime.now() - timedelta(days=1),
+            'subtype_id': self.ref('mail.mt_comment'),
+            'reply_to': False,
+            'body': 'Test follower',
+        }, {
+            'author_id': self.contact_1.id,
+            'model': 'crm.lead',
+            'res_id': self.lead_w_email.id,
+            'date': Datetime.now() - timedelta(days=20),
+            'subtype_id': self.ref('mail.mt_comment'),
+            'reply_to': False,
+            'body': 'Test follower',
+        }, {
+            'author_id': self.contact_2.id,
+            'model': 'crm.lead',
+            'res_id': self.lead_w_email.id,
+            'date': Datetime.now() - timedelta(days=15),
+            'subtype_id': self.ref('mail.mt_comment'),
+            'reply_to': False,
+            'body': 'Test follower',
+        }, {
+            'author_id': self.contact_2.id,
+            'model': 'crm.lead',
+            'res_id': self.lead_w_partner.id,
+            'date': Datetime.now() - timedelta(days=29),
+            'subtype_id': self.ref('mail.mt_comment'),
+            'reply_to': False,
+            'body': 'Test follower',
+        }, {
+            'author_id': self.contact_company.id,
+            'model': 'crm.lead',
+            'res_id': self.lead_w_email.id,
+            'date': Datetime.now() - timedelta(days=35),
+            'subtype_id': self.ref('mail.mt_comment'),
+            'reply_to': False,
+            'body': 'Test follower',
+        }, {
+            'author_id': self.contact_company.id,
+            'model': 'crm.lead',
+            'res_id': self.lead_w_partner.id,
+            'date': Datetime.now(),
+            'subtype_id': self.ref('mail.mt_comment'),
+            'reply_to': False,
+            'body': 'Test follower',
+        }])
+        initial_followers = self.lead_w_partner_company.message_follower_ids
+
+        leads = self.env['crm.lead'].browse(self.leads.ids)._sort_by_confidence_level(reverse=True)
+        master_lead = leads._merge_opportunity(max_length=None)
+        self.assertEqual(master_lead, self.lead_w_partner_company)
+
+        # Check followers of the destination lead
+        new_followers = master_lead.message_follower_ids - initial_followers
+        self.assertIn(follower_contact_2_lead_3, new_followers,
+                      'The partner must be followed the destination lead')
+        self.assertNotIn(follower_contact_1_lead_2, new_followers,
+                         'The partner was already followed the destination lead')
+        self.assertNotIn(follower_contact_2_lead_2, new_followers,
+                         'The partner was already added')
+        # "contact_company" posted a message 8 days ago on lead_2, so it's considered as inactive
```

Comment:
Would be great to have a docstring at the beginning of the test, like a small table / summary telling what are test data and what should be transfered to the new lead so that we could quickly understand what is going on in case of issue. Here it is a bit split between data and comments. Something like
```
Lead                    - Contact     - Info
lead_w_partner (MASTER) - contact_2   - OK (author message_2)
lead_w_partner (MASTER) - contact_4   - OK (author message_4)
lead_32                 - contact_3   - KO (author message_3 but old)
lead_32                 - contact_5   - KO (not author)
...
```

Reply:
good idea :)